### ISSUE TITLE
Add HAWC 2HWC to gamma-cat

### ISSUE THREAD

**Issue by cdeil** [gammapy/gamma-cat#150]
I did get the HAWC 2HWC catalog in a nice clean format. They plan to add it to the HAWC website and make it publicly available soon also. Here's the reminder issue to add it to Gammapy and links to the 2HWC catalog: https://github.com/gammapy/gammapy/issues/887#issuecomment-322992469

My 2 cents: HAWC measurements are TeV and are in scope for gamma-cat. The arguments against adding it are: 2HWC is already in a machine-readable format; it's a fixed catalog; there will be several cases in the Galactic plan where it'll be hard to decide which 2HWC "source" corresponds to an existing gamma-cat "source" and when we should create a new gamma-cat "source".

Bottom line: I think we should add it. But I'm not sure, so @GernotMaier or anyone interested - thoughts?

**Comment by pdeiml** [comment]
First of all a general question:
In gammapy there are fermi catalogs available. 
Is the idea of gamma-cat to include every source which is known today, hence, fermi and hawc as well. Or only sources within the TeV regime (this would not include fermi sources, wouldn't it)?

If the latter is the case, we should definitely add the hawc data for completeness, I think.

The correspondence to existing sources may not be such an issue because the hawc collaboration listed the TeVCat counterparts in Table 2
https://arxiv.org/pdf/1702.02992.pdf

Well, to add the data I would suggest to create one folder in input/data with the reference_id to the HAWC2 paper on arxiv, then, in that folder create tev-XXXXX.yaml files for every of the 39 sources.
In addition, to connect the existing sources to the common_names etc. we should add a new keyword '2hwc_name' in 
https://github.com/gammapy/gamma-cat/blob/master/input/schemas/basic_source_info.schema.yaml

What do you think about these ideas?

**Comment by cdeil** [comment]
> Is the idea of gamma-cat to include every source which is known today, hence, fermi and hawc as well.

It's up for discussion.

IMO we should focus on TeV and just prioritise our limited time to work on gamma-cat.
So IMO TeV measurements (individual papers) that don't exist in machine-readable format are most important, as well as scripting / webpage / paper writing for gamma-cat. Then come TeV catalogs like 2HWC and soon HGPS (HESS Galactic plane survey). And only then should we consider adding info from Fermi catalogs to gamma-cat, and in those discussions I'd probably be against adding it because too much work / too hard to maintain / too little extra value for users.

Since 2HWC is little work to add, IMO it would be OK to add even if not very high priority. (but don't be fooled, it'll still take a day for you to code and a day to review for me at least, please don't start it before the PR in Gammapy is merged). I can look at / comment on your implementation solution for gamma-cat / 2HWC if / when someone has time to implement it, from a brief look it looks good (but I don't think the choice which source is which will be simple / we might need a new scheme to declare  "sources" that aren't really independent and should not be included twice when creating the catalog.

**Comment by adonath** [comment]
@cdeil For now I'm against adding data from TeV catalogs to gamma-cat, if those are available in machine readable format. What is the point of duplicating information that is already there just in a different format? I'm not sure if gamma-cat should have the goal to be a "complete" TeV or even GeV catalog (whatever that means...). I'd even go further and say gamma-cat isn't a catalog at all, I'd rather see it as a data collection and archive (the main difference being the uniformity in the analysis of  sources...). For me, right now, gamma-cat provides access to data, that explicitly isn't available via (electronic) catalogs.   

This said, I would only continue to add information from individual source papers (of course including Fermi-LAT and HAWC publications...) and provide the possible associations with existing catalogs (such as HGPS and 2HWC) by source name, as suggested by @pdeiml.  Those could also include several sources if the association is not unique.

**Comment by cdeil** [comment]
> What is the point of duplicating information that is already there just in a different format?

To be complete.

> I'm not sure if gamma-cat should have the goal to be a "complete" TeV

E.g. Gernot thinks it's very important to have that as the goal, and I mostly agree.
Related: https://github.com/gammapy/gamma-cat/issues/66

> I'd even go further and say gamma-cat isn't a catalog at all, I'd rather see it as a data collection and archive

Gamma-cat has two goals and data products: data collection and catalog.
See https://gamma-cat.readthedocs.io/data/index.html and already the classes in `gammapy.catalog.gammacat`.

---

@adonath - I think while maybe in theory there are differences in goals / scope in this discussion, in practice we're 100% agreed: we're manpower-limited and should focus on what's needed and prioritise (but take whatever useful contributions we get). So @pdeiml - if you don't care much what you work on, I would also prefer to not add 2HWC to gamma-cat now, but first to add other data to gamma-cat, or to add 2HWC to gamma-sky.net, if you want to learn some HTML / Typescript coding.

**Comment by GernotMaier** [comment]
I am divided in this and not very helpful for the discussion

- ideal would be of course to have all information about gamma-ray sources in one place
- priority should be to make accessible what is inaccessible at this point - and these are all the HESS/MAGIC/VERITAS publications

I think this is essentially repeating what has been said above. If it is little work, let's add it. If it is more work, let's wait.
